.github/actions/setup-java-env/action.yml

```diff
@@ -0,0 +1,68 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Sets up the cache for Java dependencies. If a specific key is given, then the
+# key is used to resolve or else resolving the first entry that matches the prefix
+# without the key. Otherwise, today's date and yetserday's date (both in UTC) will be
+# tried before going to the first match without any date.
+
+name: 'Setup Java Cache'
+description: 'Sets up the Java dependency cache.'
+
+inputs:
+  cache-key:
+    type: string
+    description: 'Key to use for the cache entry'
+    required: false
+    default: ''
+  java-version:
+    type: string
+    description: 'The version of Java to install'
+    required: false
+    default: '8'
+outputs:
+  cache-hit:
+    description: 'Whether or not there was a cache hit'
+    value: ${{ steps.setup-cache.outputs.cache-hit }}
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Setup Java
+      uses: actions/setup-java@a12e082d834968c1847f782019214fadd20719f6
+      with:
+        distribution: 'zulu'
+        java-version: ${{ inputs.java-version }}
+    - name: Prepare Key
+      shell: bash
+      env:
+        KEY: ${{ inputs.key }}
+      run: |
+        if [[ "$KEY" == "" ]]; then
+          echo "TODAY=$(date -u +%Y%m%d)" >> $GITHUB_ENV
+          echo "YESTERDAY=$(date -u --date='1 day ago' +%Y%m%d)" >> $GITHUB_ENV
+        else
+          echo "TODAY=${{ inputs.key }}" >> $GITHUB_ENV
+          echo "YESTERDAY=${{ inputs.key }}" >> $GITHUB_ENV
```

Comment:
Nit: prefer `$KEY` over `${{ inputs.key }}` for consistency since we're already using it above (or just use `${{ inputs.key }}`)

Reply:
Made it more consistent. I prefer `$KEY` here, since I think it is a bit cleaner.